### ISSUE TITLE
[BUG] `no-unnecessary-arbitrary-value` does not support color object syntax

### ISSUE THREAD

**Issue by Grapedge** [francoismassart/eslint-plugin-tailwindcss#407]
**Describe the bug**
When using the [color-object-syntax](https://v3.tailwindcss.com/docs/customizing-colors#color-object-syntax), this plugin fails to correctly detect unnecessary arbitrary values.

**To Reproduce**
```javascript
var config = [
  {
    config: {
      theme: {
        colors: {
          blue: {
            50: "#abcdef",
          },
          "blue-100": "#123456",
        },
      },
    },
  },
];

ruleTester.run("arbitrary-values", rule, {
  valid: [],
  invalid: [
    {
      // AssertionError [ERR_ASSERTION]: Should have 1 error but had 0: []
      code: `<pre class="text-[#abcdef]">text-[#abcdef]</pre>`,
      options: config,
      output: `<pre class="text-blue-50">text-[#abcdef]</pre>`,
      errors: generateErrors(["text-[#abcdef]"], [["text-blue-50"]]),
    },
    {
      // This works correctly
      code: `<pre class="text-[#123456]">text-[#123456]</pre>`,
      options: config,
      output: `<pre class="text-blue-100">text-[#123456]</pre>`,
      errors: generateErrors(["text-[#123456]"], [["text-blue-100"]]),
    },
  ],
});
```

**Screenshots**

<img width="977" height="88" alt="Image" src="https://github.com/user-attachments/assets/10e2654a-c46a-436f-8366-9ef792e3a041" />

<img width="862" height="144" alt="Image" src="https://github.com/user-attachments/assets/d09fd4c9-13d4-4064-8bd1-f0b1833b51e4" />

<img width="701" height="957" alt="Image" src="https://github.com/user-attachments/assets/8f479934-bf9d-46ce-b474-0d703bc46c55" />

**Comment by lancej1022** [comment]
I might be misunderstanding this but I think my open PR https://github.com/francoismassart/eslint-plugin-tailwindcss/pull/436 would fix this. Unfortunately, I dont think the author of the repo still checks on the issues/PRs so if you try it out locally and it does fix things for you, then you might need to manually patch the package within your node_modules

**Comment by francoismassart** [comment]
Does this still concerns Tailwind CSS v4 ?
If so please make a new issue in the context of Tailwind CSS v4 as I'm dropping support for v3
